examples/filter/pipeline_passthru/main/passthru.c

```diff
@@ -0,0 +1,115 @@
+    pipeline = audio_pipeline_init(&pipeline_cfg);
+
+    ESP_LOGI(TAG, "[3.1] Create i2s stream to write data to codec chip");
+    i2s_stream_cfg_t i2s_cfg = I2S_STREAM_CFG_DEFAULT();
+    i2s_cfg.type = AUDIO_STREAM_WRITER;
+    i2s_stream_writer = i2s_stream_init(&i2s_cfg);
+
+    ESP_LOGI(TAG, "[3.2] Create i2s stream to read data from codec chip");
+    i2s_stream_cfg_t i2s_cfg_read = I2S_STREAM_CFG_DEFAULT();
+    i2s_cfg_read.type = AUDIO_STREAM_READER;
+    i2s_stream_reader = i2s_stream_init(&i2s_cfg_read);
+
+    ESP_LOGI(TAG, "[3.3] Register all elements to audio pipeline");
+    audio_pipeline_register(pipeline, i2s_stream_reader, "i2s_read");
+    audio_pipeline_register(pipeline, i2s_stream_writer, "i2s_write");
+
+    ESP_LOGI(TAG, "[3.4] Link it together [codec_chip]-->i2s_stream_reader-->i2s_stream_writer-->[codec_chip]");
+    audio_pipeline_link(pipeline, (const char *[]) {"i2s_read", "i2s_write"}, 2);
+
+    ESP_LOGI(TAG, "[ 4 ] Initialize peripherals");
+    esp_periph_config_t periph_cfg = { 0 };
+    esp_periph_init(&periph_cfg);
+
+    ESP_LOGI(TAG, "[ 5 ] Setup event listener");
+    audio_event_iface_cfg_t evt_cfg = AUDIO_EVENT_IFACE_DEFAULT_CFG();
+    audio_event_iface_handle_t evt = audio_event_iface_init(&evt_cfg);
+
+    ESP_LOGI(TAG, "[5.1] Listening event from all elements of pipeline");
+    audio_pipeline_set_listener(pipeline, evt);
+
+    ESP_LOGI(TAG, "[5.2] Listening event from peripherals");
+    audio_event_iface_set_listener(esp_periph_get_event_iface(), evt);
+
+    ESP_LOGI(TAG, "[ 6 ] Start audio_pipeline");
+    audio_pipeline_run(pipeline);
+
+    ESP_LOGI(TAG, "[ 7 ] Listen for all pipeline events");
+    while (1) {
+        audio_event_iface_msg_t msg;
+        esp_err_t ret = audio_event_iface_listen(evt, &msg, portMAX_DELAY);
+        if (ret != ESP_OK) {
+            ESP_LOGE(TAG, "[ * ] Event interface error : %d", ret); 
+            continue;
+        }
+
+        if (msg.cmd == AEL_MSG_CMD_ERROR) {
+            ESP_LOGE(TAG, "[ * ] Action command error: src_type:%d, source:%p cmd:%d, data:%p, data_len:%d",
+                     msg.source_type, msg.source, msg.cmd, msg.data, msg.data_len);
+        }
+
+        /* Stop when the last pipeline element (i2s_stream_writer in this case) receives stop event */
+        if (msg.source_type == AUDIO_ELEMENT_TYPE_ELEMENT && msg.source == (void *) i2s_stream_writer
+                && msg.cmd == AEL_MSG_CMD_REPORT_STATUS && (int) msg.data == AEL_STATUS_STATE_STOPPED) {
+            ESP_LOGW(TAG, "[ * ] Stop event received"); 
+            break;
+        }
+    }
+
+    ESP_LOGI(TAG, "[ 8 ] Stop audio_pipeline");
+    audio_pipeline_terminate(pipeline);
+
+    /* Terminate the pipeline before removing the listener */
+    audio_pipeline_remove_listener(pipeline);
+
+    /* Stop all periph before removing the listener */
+    esp_periph_stop_all();
+    audio_event_iface_remove_listener(esp_periph_get_event_iface(), evt);
```

Comment:
Finally, the above two functions calls related to peripherals should be removed as well